src/sage/libs/singular/function.pyx

```diff
@@ -1474,29 +1473,29 @@ cdef inline call_function(SingularFunction self, tuple args, object R, bint sign
 
     currentVoice = NULL
     myynest = 0
-    errorreported = 0
-
-    while error_messages:
-        error_messages.pop()
+    start_catch_error()
 
     with opt_ctx: # we are preserving the global options state here
         if signal_handler:
-            sig_on()
-            _res = self.call_handler.handle_call(argument_list, si_ring)
-            sig_off()
+            try:
+                sig_on()
+                _res = self.call_handler.handle_call(argument_list, si_ring)
+                sig_off()
+            finally:
+                s = check_error()
         else:
             _res = self.call_handler.handle_call(argument_list, si_ring)
+            s = check_error()
 
     if myynest:
         myynest = 0
```

Comment:
This looks strange: isn't this the same as just
```suggestion
    myynest = 0
```
?

But actually, I could not find any use of `myynest` in all of sage.  Do you know what it is for?

Reply:
I agree, but then the code was like that from the beginning.

Anyway it's a singular thing, from `doc/HOWTO-libsingular` in singular repo:

```
   myynest=1; /* <=0: interactive at eof/ >=1: non-interactive */
```

That's as much information as I can find. The name is pretty confusing but probably `yy` refer to Yacc.

I don't see a good reason to change it either. But the simplification looks valid, the variable is just a plain old C++ global variable.